opal/mca/common/ofi/common_ofi.c

```diff
@@ -33,6 +33,120 @@ OPAL_DECLSPEC opal_common_ofi_module_t opal_common_ofi = {.prov_include = NULL,
 
 static const char default_prov_exclude_list[] = "shm,sockets,tcp,udp,rstream,usnic";
 static opal_mutex_t opal_common_ofi_mutex = OPAL_MUTEX_STATIC_INIT;
+static bool opal_common_ofi_initialized = false;
+static int opal_common_ofi_init_ref_cnt = 0;
+
+#if OPAL_OFI_IMPORT_MONITOR_SUPPORT
+
+static int opal_common_ofi_monitor_start(struct fid_mem_monitor *monitor)
+{
+    return 0;
+}
+static void opal_common_ofi_monitor_stop(struct fid_mem_monitor *monitor)
+{
+    return;
+}
+static int opal_common_ofi_monitor_subscribe(struct fid_mem_monitor *monitor,
+                                             const void *addr, size_t len)
+{
+    return 0;
+}
+static void opal_common_ofi_monitor_unsubscribe(struct fid_mem_monitor *monitor,
+                                                const void *addr, size_t len)
+{
+    return;
+}
+static bool opal_common_ofi_monitor_valid(struct fid_mem_monitor *monitor,
+                                     const void *addr, size_t len)
+{
+    return true;
+}
+
+static struct fid_mem_monitor *opal_common_ofi_monitor;
+static struct fid *opal_common_ofi_cache_fid;
+static struct fi_ops_mem_monitor opal_common_ofi_export_ops = {
+    .size = sizeof(struct fi_ops_mem_monitor),
+    .start = opal_common_ofi_monitor_start,
```

Comment:
Instead of having these empty functions, could these just be set to null?

Reply:
I don't believe so, libfabric will call these functions even if they don't do anything, so this will segfault (I tested and that is the case).

Reply:
Got it - thanks.

Reply:
Just getting to read this PR (apologies). I too think this is odd. I think we should update libfabric at some point to check if a particular method is defined or not before invoking it, so users of the interface can just provide functions for the actions they care about.

Reply:
That makes sense, should bring that up with Sean